### PR TITLE
Transform java_library exports / transitive jar dependencies into app…

### PR DESCRIPTION
**Pull request by sconover**
…roproiate iml

add transitive dep info to debug log, include transitive_runtime_deps in iml classpath building
initial part of transitive scenario. bugfix in modules.xml creation.
add parent and child, w/ failing tests
transitive scenario
exports are visible to all downstream modules. scenario test.
notes

### CODE REVIEW

File: scenarios/04_transitive_via_export/BUILD [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package(default_visibility = ["//03_basic:__subpackages__"])
+
+load("@rules_intellij_generate//:def.bzl", "intellij_modules_xml")
```
Comment (daicoden) [inline scenarios/04_transitive_via_export/BUILD]: what does def.bzl mean?

File: scenarios/04_transitive_via_export/child/src/test/java/transitive_via_export/parent/ChildTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+class ChildTest {
+  @Test
+  public void print_hello_three_times() {
+    assertEquals("hellohello", new Child().hello2times());
```
Comment (daicoden) [inline scenarios/04_transitive_via_export/child/src/test/java/transitive_via_export/parent/ChildTest.java]: hello2times returns a boolean, not a string.

File: scenarios/04_transitive_via_export/grandparent/BUILD [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+java_test(
+  name="grandparent_tests",
+  main_class="org.junit.platform.console.ConsoleLauncher",
+  args=[
+    "--select-package transitive_via_export.grandparent",
+    "--details verbose",
+  ],
+  runtime_deps=[
+    ":grandparent_tests_lib",
+    "@org_junit_jupiter_junit_jupiter_api//jar",
```
Comment (daicoden) [inline scenarios/04_transitive_via_export/grandparent/BUILD]: this is already in :grandparent_tests_lib, is that intentional?

File: scenarios/04_transitive_via_export/BUILD (hunk 1 above)
Comment (sconover) [reply scenarios/04_transitive_via_export/BUILD]: after that thread I started on bazel-discuss, I looked around at some of the goog projects and decided to copy the style of rules_go

File: scenarios/04_transitive_via_export/grandparent/BUILD (hunk 3 above)
Comment (sconover) [reply scenarios/04_transitive_via_export/grandparent/BUILD]: `:grandparent_tests_lib` doesn't export that and I am using the api in the test code, so I have to declare this explicitly. new world :)

File: scenarios/04_transitive_via_export/child/src/test/java/transitive_via_export/parent/ChildTest.java (hunk 2 above)
Comment (sconover) [reply scenarios/04_transitive_via_export/child/src/test/java/transitive_via_export/parent/ChildTest.java]: nice catch. we don't have a build that fails yet, unfortunately :)

addressed.

**Pull request merged**